src/main/java/com/vaadin/training/bugrap/ui/BugrapApplicationModel.java

```diff
@@ -126,8 +171,46 @@ public boolean isSingleReportSelected() {
 	}
 
 	public void updateSelectedReport() {
-		BugrapFacade.getInstance().save(report);
-		UIEventBus.getInstance().publish(new ReportUpdatedEvent(report));
+		if (isSingleModificationModeSelected()) {
+			BugrapFacade.getInstance().save(report);
+			UIEventBus.getInstance().publish(new ReportUpdatedEvent(report));
+		} else {
+			for (final Report report : reports) {
+				setPropertyIfNotNull(massModificationReport::getPriority, report::setPriority);
+				setPropertyIfNotNull(massModificationReport::getType, report::setType);
+				setPropertyIfNotNull(massModificationReport::getStatus, report::setStatus);
+				setPropertyIfNotNull(massModificationReport::getAssigned, report::setAssigned);
+				setPropertyIfNotNull(massModificationReport::getVersion, report::setVersion);
+				BugrapFacade.getInstance().save(report);
+			}
+			UIEventBus.getInstance().publish(new ReportUpdatedEvent(massModificationReport));
```

Comment:
The event could take the original selected reports list as arguments. This way we wouldn't leak the artificial massModificationReport to whoever might be listening these published events.
This might also come in handy if we'd like to fine-tune the grid by updating only the selected rows instead of reinitializing the whole component whenever user updates report(s)

Reply:
Ok, great. Gonna do it.